From bcb22a3fe1391a97777a8ce57e77f0538f1def08 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Fri, 29 Sep 2023 20:40:25 +0200
Subject: [PATCH] google: add gpr lib to TTS_ENGINE call

:Release Notes:
With new grpc from:
dfac3cc512 devtools: grpc: bump to 1.56.2

  this was now failing with:
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol 'gpr_inf_future'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libgpr.so.33: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

  unfortunately this hardcodes list of libs instead of using .cmake
  which lists it correctly:
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:  IMPORTED_LOCATION_NOCONFIG "${_IMPORT_PREFIX}/lib/libgpr.so.33.0.0"
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:  IMPORTED_SONAME_NOCONFIG "libgpr.so.33"
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:list(APPEND _cmake_import_check_files_for_gRPC::gpr "${_IMPORT_PREFIX}/lib/libgpr.so.33.0.0" )

set_target_properties(gRPC::grpc PROPERTIES
  INTERFACE_COMPILE_FEATURES "cxx_std_14"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "ZLIB::ZLIB;c-ares::cares;gRPC::address_sorting;re2::re2;gRPC::upb;dl;m;Threads::Threads;rt;absl::cleanup;absl::flat_hash_map;absl::flat_hash_set;absl::inlined_vector;absl::bind_front;absl::function_ref;absl::hash;absl::type_traits;absl::statusor;absl::span;absl::utility;gRPC::gpr;OpenSSL::SSL;OpenSSL::Crypto;gRPC::address_sorting;gRPC::upb"
)

* similarly absl_synchronization:

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/src/engines/tts/google/GoogleTTSEngine.cpp.o: undefined reference to symbol '_ZN4absl12lts_202308025MutexD1Ev'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_synchronization.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080213hash_internal15MixingHashState5kSeedE'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_hash.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_202308024Cord6AppendEOS1_'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_cord.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080213cord_internal17cordz_next_sampleE'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_cordz_functions.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080212log_internal15LogMessageFatalC1EPKci'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_log_internal_message.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dircom.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080212log_internal9kCharNullE'
com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_log_internal_nullguard.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

:Detailed Notes:

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-310] Apply .patch files for hot-fix to corresponding repositories

Change-Id: I33403d27378896e2ab879899f0b4d071307a192b
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/com.webos.service.tts/+/383766 google: add gpr lib to TTS_ENGINE call]

 src/engines/tts/google/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/engines/tts/google/CMakeLists.txt b/src/engines/tts/google/CMakeLists.txt
index f04a47d..bcfdbac 100755
--- a/src/engines/tts/google/CMakeLists.txt
+++ b/src/engines/tts/google/CMakeLists.txt
@@ -26,6 +26,15 @@ set(src ${CMAKE_CURRENT_SOURCE_DIR}/GoogleTTSEngine.cpp
     ${GOOGLE_SOURCE}
 )
  set(deps
+     gpr
+     absl_city
+     absl_cord
+     absl_cordz_info
+     absl_cordz_functions
+     absl_log_internal_message
+     absl_log_internal_nullguard
+     absl_synchronization
+     absl_hash
      grpc
      grpc++
      protobuf
